src/dict.c

```diff
@@ -1218,6 +1218,112 @@ char *stringFromLongLong(long long value) {
     return s;
 }
 
+/* Evolving dict into dictX. */
+dictX *dictEvolution(dict *d) {
+    dictX *dx = zmalloc(sizeof(*dx));
+
+    /* Copy the internal constants from dict to dictX. */
+    dx->type = d->type;
+    dx->rehashidx = 0;  /* when relocating elements using. */
+    dx->pauserehash = 0;
+    dx->ht_used[0] = 0;
```

Comment:
What about initializing `dx->ht_used[1]`? Is it ok to ignore it because `reshaidx` is 0?

Reply:
yeah, looks like I did too much omitting this time.